### PR TITLE
feat: expose partitions to the lifecycle service

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

File: tests/unit/test_application.py [hunk 1]
```diff
@@ -1178,3 +1178,64 @@ def test_process_grammar_no_match(grammar_app, mocker):
...
+    assert project.parts["mypart"]["override-stage"] == dedent(
+        f"""\
+        touch {app.project_dir}/stage/default
+        touch {app.project_dir}/partitions/mypartition/stage/partition
+    """
+    )
+    assert project.parts["mypart"]["override-prime"] == dedent(
+        f"""\
+        touch {app.project_dir}/prime/default
+        touch {app.project_dir}/partitions/mypartition/prime/partition
```
Comment (lengau) [inline tests/unit/test_application.py:1239]: Good check!

**Review by lengau — APPROVED** [review]
Looks great, thanks!

File: tests/conftest.py [hunk 2]
```diff
@@ -132,13 +132,25 @@ def full_build_plan(mocker) -> list[models.BuildInfo]:
     return build_plan
 
 
+@pytest.fixture
```
Comment (lengau) [inline tests/conftest.py:135]:
```suggestion
@pytest.fixture()
```

An autoformat (`tox -f format`) should take care of that too

File: tests/unit/test_application.py (hunk 1 above)
Comment (sergiusens) [reply tests/unit/test_application.py:1239]: thanks!

**Review by mr-cal — APPROVED** [review]
Looks good

**Pull request merged**